File: src/backend_dotnet/HRMApi/Models/PointTransactionHistory.cs
using System;
using System.Collections.Generic;

namespace HRMApi.Models;

public partial class PointTransactionHistory
{
    public int Id { get; set; }

    public int EmployeeId { get; set; }

    public int Value { get; set; }

    public string Type { get; set; } = null!;

    public int? ActorId { get; set; }

    public string? Description { get; set; }

    public DateTime? CreatedAt { get; set; }

    public virtual Employee? Actor { get; set; }

    public virtual Employee Employee { get; set; } = null!;
}

File: src/backend_java/src/main/java/com/group07/human_resource_management/entity/PointTransactionHistory.java
package com.group07.human_resource_management.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "point_transaction_history")
@Getter @Setter
@Builder @NoArgsConstructor @AllArgsConstructor
public class PointTransactionHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // FK employee_id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "employee_id", nullable = false)
    private Employee employee;

    @Column(nullable = false)
    private Integer value;

    @Column(nullable = false)
    private String type; // earn, redeem, transfer, adjustment

    // actor_id → employee.id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "actor_id")
    private Employee actor;

    private String description;

    @Column(name = "created_at")
    private LocalDateTime createdAt;
}
